src/crossroads/main.py

```diff
@@ -107,3 +137,48 @@ def json_parser():
                                      longitude=(location['longitudeE7'] / 10. ** 7)))
 
     return parsed_locations
+
+def add_color_marker(parsed_locations_list, colour):
+    """
+    Adds the "marker" key and value depending on the parameter "colour" to the parsed .json file with Location History.
+
+    :param parsed_locations_list: list of dictionaries
+    :param colour: (str) the color of the marker to be added
+    :return parsed_loc_list_with_colour: a list of chronologically arranged dictionaries, from the oldest ones
+    """
+    parsed_loc_list_with_colour = parsed_locations_list
```

Comment:
You are assigning here the same data to another variable. Remember this is risky since it will mutate also original data. Generally it is better way to create another empty list and append to it modified values.